pkg/appgateserver/server.go

```diff
@@ -175,75 +175,88 @@ func (app *appGateServer) ServeHTTP(writer http.ResponseWriter, request *http.Re
 	newUrlPath, serviceId := extractServiceId(request.URL.Path)
 	request.URL.Path = newUrlPath
 
-	// Read the request body bytes.
-	requestPayloadBz, err := io.ReadAll(request.Body)
-	request.Body.Close()
+	// Create an error logger with the common fields for error logging.
+	errorLogger := app.logger.With().
+		Error().
+		Str("service_id", serviceId).
+		Fields(map[string]interface{}{
+			"method":         request.Method,
+			"url":            request.URL.String(),
+			"content_type":   request.Header.Get("Content-Type"),
+			"content_length": request.ContentLength,
+		})
+
+	poktHTTPRequest, requestBz, err := sdktypes.SerializeHTTPRequest(request)
 	if err != nil {
-		app.replyWithError(
-			ctx,
-			requestPayloadBz,
-			writer,
-			serviceId,
-			"unknown",
-			ErrAppGateHandleRelay.Wrapf("reading relay request body: %s", err),
-		)
-		// TODO_IMPROVE: log additional info?
-		app.logger.Error().Err(err).Msg("failed reading relay request body")
+		// If the request cannot not be parsed, pass an empty POKTHTTPRequest and
+		// an UNKNOWN_RPC type to the replyWithError method.
+		emptyPOKTHTTPRequest := &sdktypes.POKTHTTPRequest{}
+		rpcType := sharedtypes.RPCType_UNKNOWN_RPC
+		errorReply := ErrAppGateHandleRelay.Wrapf("parsing request: %s", err)
+
+		app.replyWithError(errorReply, emptyPOKTHTTPRequest, serviceId, rpcType, writer)
+		errorLogger.Err(err).Msg("failed parsing request")
 
 		return
 	}
+
 	app.logger.Debug().
 		Str("service_id", serviceId).
```

Comment:
Provided that we made this part of the logger, we don't have to log it every time.